pmd-visualforce/src/main/java/net/sourceforge/pmd/lang/vf/rule/security/VfUnescapeElRule.java

```diff
@@ -52,9 +47,12 @@
     private static final Pattern ON_EVENT = Pattern.compile("^on(\\w)+$");
     private static final Pattern PLACEHOLDERS = Pattern.compile("\\{(\\w|,|\\.|'|:|\\s)*\\}");
 
+    private final ElEscapeDetector escapeDetector = new ElEscapeDetector();
```

Comment:
The only changes happening in this class is the extraction of methods to a new class called ElEscapeDetector. This class helps avoid duplication of code between VfUnescapeEl and VfHtmlStyleTagXss.